File: scratch/patch-test.py
import re

with open('tests/scripts/deploy-release.test.js', 'r') as f:
    text = f.read()

text = text.replace("import fs from 'fs';", "import originalFs from 'fs';")
if "import originalFs from 'fs';" not in text:
    text = text.replace("import path from 'path';", "import path from 'path';\nimport originalFs from 'fs';")

# Replace the readFileSync mock
text = re.sub(
    r'readFileSync: mock\.fn\(\)',
    r"readFileSync: mock.fn((f) => { try { if (f.includes('tests/scripts') || f.includes('scripts/') || f.includes('node:')) return originalFs.readFileSync(f, 'utf8'); } catch (e) {} return ''; })",
    text
)

# Also fix lstatSync to fallback? Maybe not needed for stack traces.

with open('tests/scripts/deploy-release.test.js', 'w') as f:
    f.write(text)

File: scratch/patch-test.mjs
import fs from 'fs';
let code = fs.readFileSync('tests/scripts/production-preflight.test.js', 'utf8');

code = code.replace(
  /clients: \{\n\s*c1: \{\n\s*role: 'user',\n\s*projectIds: \['unknown'\]\n\s*\}\n\s*\}/,
  `clients: {\n            c1: {\n              role: 'user',\n              projectIds: ['11111111-1111-4111-8111-111111111111', 'unknown']\n            }\n          }`
);

fs.writeFileSync('tests/scripts/production-preflight.test.js', code);
